model/ftn/w3oacpmd.ftn

```diff
@@ -231,61 +230,100 @@
+                 LAT(I,1)=YGRD(IY,IX)*FACTOR
                END DO
-            END DO
-            !
-            ! areas, corners
-            ALLOCATE ( AREA(NNODES,1), CORLON(NNODES,1,4), CORLAT(NNODES,1,4) )
-            I = 0
-            DO IY = NYS, NYN
+             END DO
+             !
+             ! areas, corners
+             ALLOCATE ( AREA(NNODES,1), CORLON(NNODES,1,4), CORLAT(NNODES,1,4) )
+             I = 0
+             DO IY = NYS, NYN
                DO IX = NXW, NXE
-                  I = I+1
-                  CORLON(I,1,1)=LON(I,1)+HPFAC(IY,IX)/2.*FACTOR
-                  CORLON(I,1,2)=LON(I,1)-HPFAC(IY,IX)/2.*FACTOR
-                  CORLON(I,1,3)=LON(I,1)-HPFAC(IY,IX)/2.*FACTOR
-                  CORLON(I,1,4)=LON(I,1)+HPFAC(IY,IX)/2.*FACTOR
-                  CORLAT(I,1,1)=LAT(I,1)+HQFAC(IY,IX)/2.*FACTOR
-                  CORLAT(I,1,2)=LAT(I,1)+HQFAC(IY,IX)/2.*FACTOR
-                  CORLAT(I,1,3)=LAT(I,1)-HQFAC(IY,IX)/2.*FACTOR
-                  CORLAT(I,1,4)=LAT(I,1)-HQFAC(IY,IX)/2.*FACTOR
-                  AREA(I,1)=HPFAC(IY,IX)*HQFAC(IY,IX)
+                 I = I+1
+                 CORLON(I,1,1)=LON(I,1)+HPFAC(IY,IX)/2.*FACTOR
+                 CORLON(I,1,2)=LON(I,1)-HPFAC(IY,IX)/2.*FACTOR
+                 CORLON(I,1,3)=LON(I,1)-HPFAC(IY,IX)/2.*FACTOR
+                 CORLON(I,1,4)=LON(I,1)+HPFAC(IY,IX)/2.*FACTOR
+                 CORLAT(I,1,1)=LAT(I,1)+HQFAC(IY,IX)/2.*FACTOR
+                 CORLAT(I,1,2)=LAT(I,1)+HQFAC(IY,IX)/2.*FACTOR
+                 CORLAT(I,1,3)=LAT(I,1)-HQFAC(IY,IX)/2.*FACTOR
+                 CORLAT(I,1,4)=LAT(I,1)-HQFAC(IY,IX)/2.*FACTOR
+                 AREA(I,1)=HPFAC(IY,IX)*HQFAC(IY,IX)
                END DO
-            END DO
-            !
-            ! Model grid mask
-            ALLOCATE ( MASK(NNODES,1) )
-            I = 0
-            DO IY = NYS, NYN
+             END DO
+             !
+             ! Model grid mask
+             ALLOCATE ( MASK(NNODES,1) )
+             I = 0
+             DO IY = NYS, NYN
                DO IX = NXW, NXE
-                  I = I+1
-                  ! Get the mask : 0 - sea  / 1 - open boundary cells (the land is already excluded)
-                  IF ((MAPSTA(IY,IX) .EQ. 1)) THEN
-                    MASK(I,1) = 0
-                  ELSE
-                    MASK(I,1) = 1
-                  END IF
+                 I = I+1
+                 ! Get the mask : 0 - sea  / 1 - open boundary cells (the land is already excluded)
+                 IF ((MAPSTA(IY,IX) .EQ. 1)) THEN
+                   MASK(I,1) = 0
+                 ELSE
+                   MASK(I,1) = 1
+                 END IF
                END DO
-            END DO
+             END DO
+!/SMC           ELSE
+!/SMC             !
+!/SMC             ! 1.2. SMC grids
+!/SMC             ! ----------------------------------
+!/SMC             NNODES = NSEA
+!/SMC             !
+!/SMC             ! Distinguish between 2 and 4-level SMC models
```

Comment:
Only a small thing:
I think this comment is a bit misleading, the DLON and DLAT values are the smallest grid cell increments (i.e. the tier 1 cells). The number of tiers could (theoretically!) be set to any value, although 2, 3 or 4 are the likely to be the most common.